addons/account_check_printing/models/account_payment.py

```diff
@@ -80,13 +60,13 @@ def _check_communication(self, payment_method_id, communication):
             if len(communication) > 60:
                 raise ValidationError(_("A check memo cannot exceed 60 characters."))
 
-    @api.model
-    def create(self, vals):
-        if vals['payment_method_id'] == self.env.ref('account_check_printing.account_payment_method_check').id\
-                and vals.get('check_manual_sequencing'):
-            sequence = self.env['account.journal'].browse(vals['journal_id']).check_sequence_id
-            vals.update({'check_number': sequence.next_by_id()})
-        return super(AccountPayment, self).create(vals)
+    @api.multi
+    def post(self):
+        super(AccountPayment, self).post()
+        payment_method_check = self.env.ref('account_check_printing.account_payment_method_check')
+        for payment in self.filtered(lambda r: r.payment_method_id == payment_method_check and r.check_manual_sequencing):
+            sequence = payment.journal_id.check_sequence_id
+            payment.check_number = sequence.next_by_id()
```

Comment:
You should return the result of `super`
`res = super(AccountPayment, self).post()`
`...`
`return res`